### DISCUSSION TITLE
Callback URL

### DISCUSSION THREAD

**Discussion by collo54** [serverpod/serverpod#2140] [Q&A] [answered]
Is it possible to create a callback url with serverpod. This is used in to receive results from another Third party API. Also is there a difference between webhooks and callback URLs in serverpod.

**Comment by Isakdl** [comment, marked as answer]
Hey @collo54 !

In this context, a webhook and callback URL are the same thing.

If you need to conform to a specific layout of the request the easiest thing is to use the webserver in serverpod. This is an experimental feature right now and not too well documented. https://docs.serverpod.dev/concepts/webserver


Here is an example of how you could use it, you can extend the `Route` class to handle the request yourself.

```dart
class ExampleRoute extends Route {
  @override
  Future<bool> handleCall(Session session, HttpRequest request) async {
    // Http request uri
    request.uri;

    // Http request method GET, POST, PUT, DELETE etc...
    request.method;

    // Http request headers
    request.headers;

    // Http request body, currently only supports form requests (I.E. the header `Content-Type` set to `application/x-www-form-urlencoded`)
    Map<String, String> body = await Route.getBody(request);

    // Write data as a response
    request.response.write({});

    // Close the response
    await request.response.close();

    // Return true if this method handled the request otherwise false
    return true;
  }
}

```

Then inside your `server.dart` file you can add the route to the web server like so:

```dart
// server.dart
void run(List<String> args) async {
  // Initialize Serverpod and connect it with your generated code.
  final pod = Serverpod(
    args,
    Protocol(),
    Endpoints(),
  );

  ...

  pod.webServer.addRoute(ExampleRoute(), '/example');

  ...

  await pod.start();
}
```

In dev mode, you can then access this route by calling `http://localhost:8082/example` (the web server is serving traffic on 8082 by default).
